mqtt/mqtt-broker/src/settings/mod.rs

```diff
@@ -145,13 +145,34 @@ impl Default for RetainedMessagesConfig {
     }
 }
 
-// TODO: apply settings
 #[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct SessionPersistenceConfig {
     file_path: String,
     #[serde(with = "humantime_serde")]
     time_interval: Duration,
-    unsaved_message_count: u32,
+}
+
+impl SessionPersistenceConfig {
+    pub fn new(file_path: String, time_interval: Duration) -> Self {
+        Self {
+            file_path,
+            time_interval,
+        }
+    }
+
+    pub fn file_path(&self) -> String {
+        self.file_path.clone()
+    }
+
+    pub fn time_interval(&self) -> Duration {
+        self.time_interval
+    }
+}
+
+impl Default for SessionPersistenceConfig {
```

Comment:
When doing so we make defaults scattered across multiple places. Let's leave it in default.json only? If you see more defaults for the settings, please remove it as well.

Reply:
Aren't these defaults needed for this?
https://github.com/Azure/iotedge/blob/iiot/mqtt/mqtt-broker/src/broker.rs#L944

I think we need to have BrokerConfig default to reading the appropriate config.